wisl/examples/SLL_ex_ongoing.wisl

```diff
@@ -0,0 +1,160 @@
+//
+//  SLL predicates
+//
+
+// With addresses
+predicate SLL_addrs(+x, xs : List) {
+  // Empty SLL
+  (x == null) * (xs == []);
+  // One SLL node and the rest
+  (x -b> #v, #y) * SLL_addrs(#y, #xs) * (xs == #y :: #xs) [ox: #v]
+}
+
+// With values
+predicate SLL_vals(+x, vs : List) {
+  // Empty SLL
+  (x == null) * (vs == []);
+  // One SLL node and the rest
+  (x -b> #v, #y) * SLL_vals(#y, #vs) * (vs == #v :: #vs) [ox: #y]
+}
+
+// With values
+predicate SLL(+x, xs : List, vs : List) {
+  // Empty SLL
+  (x == null) * (xs == []) * (vs == []);
+  // One SLL node and the rest
+  (x -b> #v, #y) * SLL(#y, #xs, #vs) * (xs == #y :: #xs) * (vs == #v :: #vs) [ox: ]
+}
+
+// With length
+predicate SLL_len(+x, n : Int) {
+  // Empty SLL
+  (x == null) * (n == 0);
+  // One SLL node and the rest
+  (x -b> #v, #y) * (0 <# n) * SLL_len(#y, n - 1) [ox: #v, #y]
+}
+
+//
+//
+//  Examples
+//
+//
+
+// General pre-condition of list length
+predicate SLL_length_pre(+x, def, var : Int) {
+    (def == 0) * (x == #x) * SLL_addrs(#x, #xs) * (var == len #xs);
+    (def == 1) * (x == #x) * SLL_vals(#x, #vs) * (var == len #vs);
+    (def == 2) * (x == #x) * SLL(#x, #xs, #vs) * (var == len #xs);
+    (def == 3) * (x == #x) * SLL_len(#x, #n) * (var == #n)
```

Comment:
Note for future, this strongly look like tags
What could be added in Gillian in the future there, so that this looks prettier and probably more efficient?

Reply:
Predicate definition tags? How does that look like? They'd need to be visible in order to establish the connection between the pre-condition and the post-condition.